genai.go

```diff
@@ -324,6 +341,106 @@ type AgentAPIKeyUpdateRequest struct {
 	Name       string `json:"name,omitempty"`
 }
 
+type KnowledgeBaseCreateRequest struct {
+	DatabaseID         string                    `json:"database_id"`
+	DataSources        []KnowledgeBaseDataSource `json:"datasources"`
+	EmbeddingModelUUID string                    `json:"embedding_model_uuid"`
+	Name               string                    `json:"name"`
+	ProjectID          string                    `json:"project_id"`
+	Region             string                    `json:"region"`
+	Tags               []string                  `json:"tags"`
+	VPCUUID            string                    `json:"vpc_uuid"`
+}
+
+// KnowledgeBaseDataSource represents a Gen AI Knowledge Base Data Source
+type KnowledgeBaseDataSource struct {
+	BucketName           string                `json:"bucket_name,omitempty"`
```

Comment:
It's not super clear from the docs, but it appears bucket_name and region are deprecated and moved to `spaces_date_source`:
<img width="835" alt="Screenshot 2025-06-02 at 10 58 33 AM" src="https://github.com/user-attachments/assets/6090f8f6-30a6-415a-9d17-6ea00e376945" />